fw/http_frame.c

```diff
@@ -742,11 +750,15 @@ tfw_h2_stream_id_close(TfwHttpReq *req, unsigned char type,
 		id = stream->id;
 	}
 
+	if (type < _HTTP2_UNDEFINED && !(flags & HTTP2_F_END_STREAM))
+		goto done;
+
```

Comment:
I don't understand the change, especially how does it affect non-websocket HTTP/2 streams. Why do we need the change? Maybe it's better to discuss on a call.

Anyway, please merge this condition with the previous one under single `if (type < _HTTP2_UNDEFINED)`.

This isn't about this patch set, but each call of the function (and we have a lot of calls, even in cache.c) to get id of the stream calls `STREAM_SEND_PROCESS()` -> `tfw_h2_stream_fsm()` - the stream FSM. Why should we execute the FSM just to get the stream id?

Reply:
Merged with previous condition.

It was needed because for http2 websocket bootstrap we should not close stream on response (end of websocket handshake). Stream closed only on end of websocket communication.